src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st

```diff
@@ -105,12 +87,61 @@ ClyBrowserMorph >> confirmUnusedVariablesInDefiningClass: variables [
 ClyBrowserMorph >> decorateMethodEditor: aMethodEditor [
 ]
 
+{ #category : #'*Calypso-SystemTools-Core' }
+ClyBrowserMorph >> hasReferencesToClasses: classes [
+	
+	| query references |
+	
+	query := (ClyClassReferencesQuery toAny: classes from: self systemScope).
+	references := query execute.	
+	^(references notEmpty and: [
+		((references items collect: [:each | each origin])
+			difference: (classes flatCollect: [ :each | { each . each classSide } ])) notEmpty])
+]
+
 { #category : #'*Calypso-SystemTools-Core' }
 ClyBrowserMorph >> isMethodSelected: aMethod [
 
 	^self methodSelection includesActualObject: aMethod
 ]
 
+{ #category : #'*Calypso-SystemTools-Core' }
+ClyBrowserMorph >> removeClassStrategiesFor: classes [
+
+	 | hasReferences hasSubclasses hasUsers |
+	 hasReferences := self hasReferencesToClasses: classes.
+	 hasSubclasses := ((classes flatCollect: [ :each | each subclasses ]) 
+		                   copyWithoutAll: classes) notEmpty.
+	 hasUsers := ((classes flatCollect: [ :each | each users ]) 
+		              copyWithoutAll: classes) notEmpty.
+
+	 hasReferences | hasSubclasses | hasUsers ifFalse: [ 
+		 ^ {SycSilentlyRemoveClassStrategy new} ].
+
```

Comment:
Cool that you follow the method removal approach. 
Now with your nice changes and tests I think this logic for references analysis can be moved to SycRemoveClassStrategy (and same refactoring for method removal logic).
(some thoughts for possible improvements)